relations/src/r1cs/constraint_system.rs

```diff
@@ -481,10 +512,19 @@ impl<F: Field> ConstraintSystem<F> {
     /// Useful for SNARKs like [\[Marlin\]](https://eprint.iacr.org/2019/1047) or
     /// [\[Fractal\]](https://eprint.iacr.org/2019/1076), where addition gates
     /// are not cheap.
-    pub fn reduce_constraint_weight(&mut self) {
+    fn reduce_constraint_weight(&mut self) {
         self.outline_lcs();
     }
```

Comment:
We can remove this now, right?

Reply:
Wasn't the idea to add other weight-reducing optimizations, apart from just outlining? (That's what the comment here says.) In which case we'd still want this as a separate function